File: stress/t/arr_histogram.c
#include <stdio.h>
#include <stdint.h>
int main(void){
  int64_t i,n,k,v,idx,t,total,mx;
  int64_t A[40],H[7];
  n=40; k=7;
  for(i=0;i<k;i++){ H[i]=0; }
  for(i=0;i<n;i++){ A[i]=i*i*3+11; }
  for(i=0;i<n;i++){ v=A[i]; idx=v%k; t=H[idx]; H[idx]=t+1; }
  total=0; mx=0;
  for(i=0;i<k;i++){
    t=H[i]; total=total+t;
    if(t>mx){ mx=t; }
    printf("h"); printf("%ld",(long)i); printf("="); printf("%ld",(long)t); printf("\n");
  }
  printf("total="); printf("%ld",(long)total); printf("\n");
  printf("maxbin="); printf("%ld",(long)mx); printf("\n");
  return 0;
}

File: stress/t/arr_histogram.f
      PROGRAM ARRHIST
      INTEGER*8 I,N,K,V,IDX,T,TOTAL,MX
      INTEGER*8 A(0:39),H(0:6)
      N=40
      K=7
      I=0
   10 IF (I.GE.K) GOTO 20
      H(I)=0
      I=I+1
      GOTO 10
   20 CONTINUE
      I=0
   30 IF (I.GE.N) GOTO 40
      A(I)=I*I*3+11
      I=I+1
      GOTO 30
   40 CONTINUE
      I=0
   50 IF (I.GE.N) GOTO 60
      V=A(I)
      IDX=MOD(V,K)
      T=H(IDX)
      H(IDX)=T+1
      I=I+1
      GOTO 50
   60 CONTINUE
      TOTAL=0
      MX=0
      I=0
   70 IF (I.GE.K) GOTO 80
      T=H(I)
      TOTAL=TOTAL+T
      IF (T.GT.MX) MX=T
      WRITE(*,'(A,I0,A,I0)') 'h',I,'=',T
      I=I+1
      GOTO 70
   80 CONTINUE
      WRITE(*,'(A,I0)') 'total=',TOTAL
      WRITE(*,'(A,I0)') 'maxbin=',MX
      END
